packages/ember-data/lib/serializers/json-serializer.js

```diff
@@ -158,13 +158,14 @@ export default Serializer.extend({
   */
   normalize: function(type, hash) {
     if (!hash) { return hash; }
+    var model = this.store.modelFor(type);
```

Comment:
While we are at it, can we maybe figure out a better name than model for the actual class. Something like using type & typeKey or typeClass. Model implies an actual record, very easy to get confused